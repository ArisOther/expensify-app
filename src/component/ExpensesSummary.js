import React from 'react'
import {connect} from 'react-redux'
import numeral from 'numeral'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'

//logic untuk menentukan dna menampilkan count, kata, dan total
export const ExpensesSummary = ({expenseCount, expensesTotal}) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses' //jika expensecount = 1 maka kalimatnya 'expense', jika tidak maka 'expenses'
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00')
    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
        </div>
    )
}

//menghubungkan ke redux store
const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)

    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)