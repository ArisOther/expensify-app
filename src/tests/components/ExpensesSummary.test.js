import React from 'react'
import {shallow} from 'enzyme'
import ExpensesSummary from '../../component/ExpensesSummary'



test('harus bisa merender ExpensesSummary dengan data 1 expense', ()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>)
    expect(wrapper).toMatchSnapshot()
})

test('harus bisa merender ExpensesSummary dengan data multiple expense', ()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={23512340987}/>)
    expect(wrapper).toMatchSnapshot()
})