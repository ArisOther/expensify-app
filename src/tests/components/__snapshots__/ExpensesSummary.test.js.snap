// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`harus bisa merender ExpensesSummary dengan data 1 expense 1`] = `
<div>
  <h1>
    1
     
    expense
     totalling 
    $2.35
  </h1>
  viewing
</div>
`;

exports[`harus bisa merender ExpensesSummary dengan data multiple expense 1`] = `
<div>
  <h1>
    23
     
    expenses
     totalling 
    $235,123,409.87
  </h1>
  viewing
</div>
`;
